src/osd/osd_types.h

```diff
@@ -506,14 +516,12 @@ class coll_t {
     TYPE_LEGACY_TEMP = 1,  /* no longer used */
     TYPE_PG = 2,
     TYPE_PG_TEMP = 3,
-    TYPE_PG_REMOVAL = 4,   /* note: deprecated, not encoded */
   };
   type_t type;
   spg_t pgid;
   uint64_t removal_seq;  // note: deprecated, not encoded
 
-  string _str;  // cached string
-
+  char _str[spg_t::calc_name_buf_size];
```

Comment:
@liewegas I don't know what you meant here. char _str is a placeholder for whole pg name (max int, max hex, suffixes, '\0' etc) and it exists in coll_t where the name is generated and returned. For spg_t is even bigger by 1 that it should be (I think 's' suffix excludes 'p').

pg_t:;calc_name_buf_size  and spg_t::calc_name_buf_size are separate because there are operators that use char _str too, inside them size matches the max pg name length.